vmdb/app/controllers/pxe_controller/iso_datastores.rb

```diff
@@ -286,10 +285,9 @@ def iso_datastore_set_form_vars
     @edit[:key] = "isd_edit__#{@isd.id || "new"}"
     @edit[:rec_id] = @isd.id || nil
     @edit[:new][:ems_id] = @isd.ext_management_system ? @isd.ext_management_system.id : nil
-    @edit[:emses] = Array.new
-    EmsRedhat.find(:all).delete_if{|e| e.iso_datastore != nil}.sort{|a,b| a.name <=> b.name}.each do |ems|
-      @edit[:emses].push([ems.name,ems.id])
-    end
+
+    emses_with_iso_datastores = EmsRedhat.includes(:iso_datastore).select(&:iso_datastore)
+    @edit[:emses] = emses_with_iso_datastores.sort_by(&:name).collect { |ems| [ems.name, ems.id] }
```

Comment:
I could, but I'd rather not write SQL snippets either...could Arelize, but again, separate PR...

I personally think none of this logic should be in the controllers anyway...they should be model methods with tests.